target/linux/generic/hack-5.15/766-net-phy-realtek-add-led-link-select-for-RTL8221.patch

```diff
@@ -0,0 +1,69 @@
+From 4c6b89509ddaccac270a4fac9277b67eedd09b83 Mon Sep 17 00:00:00 2001
+From: Yangyu Chen <cyy@cyyself.name>
+Date: Sat, 22 Apr 2023 19:01:18 +0800
+Subject: [PATCH] net: phy: realtek: add led-link-select for RTL8221
+
+RTL8221B PHYs will select the different speeds for 3 LEDs to 10M/100M/1G
+respectively by default. Some devices like TP-LINK XDR-6088/XDR-6086
+have only one LED connects to the PHY chip LED0. If we didn't change the
+default settings, the LED will only blink at 10M speed.
+
+This patch allows configuring LED link select bitmask from DT. And it
+has been tested with TP-Link XDR6088 with different DT configurations.
+
+Signed-off-by: Yangyu Chen <cyy@cyyself.name>
+---
+ drivers/net/phy/realtek.c | 28 +++++++++++++++++++++++++++-
+ 1 file changed, 27 insertions(+), 1 deletion(-)
+
+--- a/drivers/net/phy/realtek.c
++++ b/drivers/net/phy/realtek.c
+@@ -66,6 +66,10 @@
+ #define RTL8221B_PHYCR1_ALDPS_EN		BIT(2)
+ #define RTL8221B_PHYCR1_ALDPS_XTAL_OFF_EN	BIT(12)
+ 
++#define RTL8221B_NR_LEDS				3
++#define RTL8221B_LED_LINK_SELECT		0xd032
++#define RTL8221B_LED_LINK_SELECT_OFFSET	0x2
++
+ #define RTL8366RB_POWER_SAVE			0x15
+ #define RTL8366RB_POWER_SAVE_ON			BIT(12)
+ 
+@@ -885,6 +889,28 @@ static irqreturn_t rtl9000a_handle_inter
+ 	return IRQ_HANDLED;
+ }
+ 
++static int rtl8221b_config_led(struct phy_device *phydev) {
++	struct device *node = &phydev->mdio.dev;
++	u32 link_select[RTL8221B_NR_LEDS];
++	int i, val;
++
++	val = device_property_read_u32_array(node, "realtek,led-link-select", 
++			link_select, RTL8221B_NR_LEDS);
++
++	if (val) {
++		return 0;
++	}
```

Comment:
Please remove the brackets if there is only one statement

Reply:
> Please remove the brackets if there is only one statement

fixed